plugins/lsp/schemastore.nix

```
@@ -0,0 +1,151 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+helpers.vim-plugin.mkVimPlugin config {
+  name = "schemastore";
+  originalName = "SchemaStore.nvim";
+  defaultPackage = pkgs.vimPlugins.SchemaStore-nvim;
+
+  maintainers = [ maintainers.GaetanLepage ];
+
+  extraOptions =
+    let
+      schemaOpts =
+        let
+          schemaEntry = types.submodule {
+            options = {
+              name = mkOption {
+                type = types.str;
+                description = "The name of the schema.";
+              };
+
+              description = helpers.mkNullOrStr "A description for this schema.";
+
+              fileMatch = helpers.mkNullOrOption (with types; either str (listOf str)) ''
+                Which filename to match against for this schema.
+              '';
+
+              url = mkOption {
+                type = types.str;
+                description = "The URL of this schema.";
+              };
+            };
+          };
+        in
+        {
+          select = helpers.defaultNullOpts.mkListOf types.str "[]" ''
```

Update to new non-string defaults:

```suggestion
          select = helpers.defaultNullOpts.mkListOf types.str [ ] ''
```